icu4c/source/common/brkiter.cpp

```diff
@@ -496,20 +504,25 @@ BreakIterator::makeInstance(const Locale& loc, int32_t kind, UErrorCode& status)
 
 Locale
 BreakIterator::getLocale(ULocDataLocaleType type, UErrorCode& status) const {
+    if (U_FAILURE(status)) {
+        return Locale::getRoot();
+    }
     if (type == ULOC_REQUESTED_LOCALE) {
-        return {requestLocale};
+        return requestLocale == nullptr ?
+            Locale::getRoot() : Locale(requestLocale->data());
     }
-    U_LOCALE_BASED(locBased, *this);
-    return locBased.getLocale(type, status);
+    return LocaleBased::getLocale(validLocale, actualLocale, type, status);
 }
 
 const char *
 BreakIterator::getLocaleID(ULocDataLocaleType type, UErrorCode& status) const {
+    if (U_FAILURE(status)) {
+        return nullptr;
+    }
     if (type == ULOC_REQUESTED_LOCALE) {
-        return requestLocale;
+        return requestLocale == nullptr ?  nullptr : requestLocale->data();
```

Comment:
avoid behavior change; non-error return used to never be nullptr
```suggestion
        return requestLocale == nullptr ?  "" : requestLocale->data();
```